optuna/samplers/nsgaii/_constraints_evaluation.py

```diff
@@ -86,15 +85,45 @@ def _constrained_dominates(
     return violation0 < violation1
 
 
+def _evaluate_penalty(population: Sequence[FrozenTrial]) -> np.ndarray:
+    """Evaluate feasibility of trials in population.
+    Returns:
+        A list of feasibility status T/F/None of trials in population, where T/F means
+        feasible/infeasible and None means that the trial does not have constraint values.
+    """
+
+    penalty: list[float] = []
+    for trial in population:
+        constraints = trial.system_attrs.get(_CONSTRAINTS_KEY)
+        if constraints is None:
+            penalty.append(np.nan)
+        else:
+            assert isinstance(constraints, (list, tuple))
```

Comment:
The type of the constraints is `Sequence[float]` by [definition,](https://github.com/optuna/optuna/blob/master/optuna/samplers/nsgaii/_sampler.py#L142) which includes not only list and tuple but also range and concrete classes inheriting from `collections.abc.Sequence`. Can we avoid this assertion?

Reply:
This assertion came from here:
https://github.com/optuna/optuna/blob/53d85542ae8e291101e929ceb4bbbdb146609613/optuna/samplers/nsgaii/_dominates.py#L56-L57
but seems unnecessary. I'll remove it.